File: client-app/src/app/models/order/ReserveRequest.ts
export interface ReserveRequest {
    // -----------------------------------------------------------------
    // Primary key
    // -----------------------------------------------------------------
    reserveRequestId: string;

    // -----------------------------------------------------------------
    // Foreign keys
    // -----------------------------------------------------------------
    productId: string;
    fromPartyId: string;        // Customer
    employeePartyId?: string | null;

    // -----------------------------------------------------------------
    // Business fields
    // -----------------------------------------------------------------
    reserveDate?: Date | string | null;     // API may return string; handleDatesArray will convert
    reserveAmount?: number | null;
    comments?: string | null;
    payMethod?: string | null;
    statusId?: string | null;

    // -----------------------------------------------------------------
    // Audit stamps
    // -----------------------------------------------------------------
    lastUpdatedStamp?: Date | string | null;
    createdStamp?: Date | string | null;

    // -----------------------------------------------------------------
    // Navigation properties (populated by API if included)
    // -----------------------------------------------------------------
    status?: StatusItem | null;
    product?: Product | null;
    customer?: Party | null;                // FromParty (customer)
    employee?: Party | null;                // Employee who handled the request

    // -----------------------------------------------------------------
    // Optional flattened fields (commonly added by API for grid display)
    // -----------------------------------------------------------------
    // REFACTOR: Added commonly used denormalized fields for easier grid binding
    // Purpose: Avoid deep navigation (e.g., product?.apartmentName) in Kendo Grid if API flattens them
    // Context: Adjust or remove based on what your backend actually returns
    apartmentName?: string | null;
    projectName?: string | null;
    fromPartyName?: string | null;          // Customer name
    employeeName?: string | null;
    statusDescription?: string | null;
}

File: Domain/ReserveRequest.cs
namespace Domain;

public class ReserveRequest
{
    public ReserveRequest()
    {
        // No child collections required for now
    }

    // -----------------------------------------------------------------
    // Primary key
    // -----------------------------------------------------------------
    public string ReserveRequestId { get; set; } = null!;

    public string ProductId { get; set; } = null!;
    public string FromPartyId { get; set; } = null!;

    public DateTime? ReserveDate                     { get; set; }
    public decimal?  ReserveAmount                   { get; set; }
    public string?   Comments                     { get; set; }
    public string?   PayMethod                     { get; set; }
    public string?   ChequeStatus                     { get; set; }
    public string? StatusId { get; set; }
    public string? EmployeePartyId { get; set; }


    // -----------------------------------------------------------------
    // Audit stamps (same pattern as Product)
    // -----------------------------------------------------------------
    public DateTime? LastUpdatedStamp  { get; set; }
    public DateTime? CreatedStamp      { get; set; }

    // -----------------------------------------------------------------
    // Navigation properties
    // -----------------------------------------------------------------
    public virtual StatusItem? Status { get; set; }
    public virtual Product   Product   { get; set; } = null!;
    public virtual Party     Customer  { get; set; } = null!;
    public virtual Party? Employee { get; set; }      // navigation for the employee

}